File: src/Sort 0s, 1s and 2s/Code/solution.cpp
class Solution
{
public:
    void sort012(vector<int> &arr)
    {
        int low = 0, mid = 0;
        int high = arr.size() - 1;

        // Traverse the array till mid crosses high
        while (mid <= high)
        {
            if (arr[mid] == 0)
            {
                swap(arr[low], arr[mid]);
                low++;
                mid++;
            }
            else if (arr[mid] == 1)
            {
                mid++;
            }
            else
            { // arr[mid] == 2
                swap(arr[mid], arr[high]);
                high--;
            }
        }
    }
};

File: src/Sort 0s, 1s and 2s/Code/solution.js
class Solution {
  sort012(arr) {
    let low = 0,
      mid = 0;
    let high = arr.length - 1;

    while (mid <= high) {
      if (arr[mid] === 0) {
        [arr[low], arr[mid]] = [arr[mid], arr[low]];
        low++;
        mid++;
      } else if (arr[mid] === 1) {
        mid++;
      } else {
        // arr[mid] === 2
        [arr[mid], arr[high]] = [arr[high], arr[mid]];
        high--;
      }
    }
  }
}
